system/lib/libc/musl/src/time/__tz.c

```diff
@@ -20,7 +20,8 @@ const char __gmt[] = "GMT";
 static int dst_off;
 static int r0[5], r1[5];
 
-static const unsigned char *zi, *trans, *index, *types, *abbrevs, *abbrevs_end;
+// XXX EMSCRIPTEN: index conflicts with the function index(), rename to _index.
+static const unsigned char *zi, *trans, *_index, *types, *abbrevs, *abbrevs_end;
```

Comment:
what function is this? Why should a static variable conflict? Is this just in fastcomp where we include the whole bc file?

Reply:
The conflict is with the [`index()` function](https://linux.die.net/man/3/index) which is included into here.

It seems like in theory a compiler could ignore such a conflict (especially since it's a `static` local in this file), but clang errors on it...

Reply:
What kind of error? Because this var is static, IIUC it should end up as a private symbol and should not cause multiple definition errors. So if we are getting an error, it suggests that something is wrong with either how we are building our libs, or with the linker's symbol resolution.

Reply:
Here's a simplified testcase:
```c
#include <string.h>

static const unsigned char *index;
```
With plain upstream clang that errors with
```bash
$ ~/Dev/llvm/build/bin/clang a.c
a.c:3:29: error: redefinition of 'index' as different kind of symbol
static const unsigned char *index;
                            ^
/usr/include/strings.h:68:14: note: previous definition is here
extern char *index (const char *__s, int __c)
             ^
1 error generated.
```
Using gcc or emcc leads to similar errors.

This does seem odd to me too, but it seems like the standard behavior..?

Reply:
Full emcc error (maybe the path matters?) is
```bash
$ ./emcc a.c
a.c:3:29: error: redefinition of 'index' as different kind of symbol
static const unsigned char *index;
                            ^
/usr/local/google/home/azakai/Dev/emscripten/system/include/libc/strings.h:19:7: note: previous definition is here
char *index (const char *, int);
      ^
1 error generated.
```

Reply:
I took a look at why this wasn't breaking upstream in musl.   It seems to be related to when "strrings.h" gets included by "string.h" .  `index` is defined in `strings.h`, but `src/time/__tz.c` includes only `string.h`.

It looks like the problem is that musl expects to built without `_BSD_SOURCE` but its seems to be defined when we build it in emscripten.     The solution seem to be to add `-D_XOPEN_SOURCE=700` to the cflags when building musl.   musl does this in its Makefile, we should do the same

Reply:
Interesting, thanks! Yes, I confirmed that this fixes it. I opened https://github.com/emscripten-core/emscripten/pull/9518 for that.